src/EasyAbp.Abp.PhoneNumberLogin.Application/EasyAbp/Abp/PhoneNumberLogin/Account/PhoneNumberLoginAccountAppService.cs

```diff
@@ -276,7 +296,19 @@ protected virtual async Task<string> GenerateCodeAsync(string phoneNumber, Verif
             switch (type)
             {
                 case VerificationCodeType.ResetPassword:
-                    code = await _identityUserManager.GeneratePasswordResetTokenAsync(identityUser);
```

Comment:
Why not use ASP.NET Core's `IdentityUserManager.GeneratePasswordResetTokenAsync();`? The default implementation of IUserTwoFactorTokenProvider<TUser> is fine and EasyAbp also provides one: https://github.com/EasyAbp/Abp.VerificationCode/blob/master/src/EasyAbp.Abp.VerificationCode.Identity/EasyAbp/Abp/VerificationCode/Identity/AbpVerificationCodeTokenProvider.cs.

Reply:
GeneratePasswordResetTokenAsync() this method generate code length too long , the sms  not allow code longer than 24 in aliyun .

Reply:
You can replece the default implementation, see https://github.com/EasyAbp/Abp.VerificationCode/blob/master/src/EasyAbp.Abp.VerificationCode.Identity/EasyAbp/Abp/VerificationCode/Identity/AbpVerificationCodeTokenProvider.cs

Reply:
the Abp.VerficationCode not implement twoFactor methods yet . I need to take time to fix that repository ,wait some days please when i have time .